src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -738,6 +738,60 @@ def has_pending_clones(self, snapname):
                 return False
             raise
 
+    def add_orphan_clones(self, track_id, snapname):
+        self.metadata_mgr.add_section('orphan clones')
+        self.metadata_mgr.update_section('orphan clones', track_id, snapname)
+        self.metadata_mgr.flush()
+
+    def get_pending_clones(self, snapname):
+        pending_clones = {}  # type: Dict[str, Any]
+        pending_track_id_list = []
+        pending_clone_list = []
+        index_path = ""
+
+        try:
+            if self.has_pending_clones(snapname):
+                pending_track_id_list = self.metadata_mgr.list_all_keys_with_specified_values_from_section('clone snaps', snapname)
+        except MetadataMgrException as me:
+            if me.errno != -errno.ENOENT:
+                raise
+
+        try:
+            with open_clone_index(self.fs, self.vol_spec) as index:
+                index_path = index.path.decode('utf-8')
+        except IndexException as e:
+            log.warning("failed to open clone index: {0}".format(e))
```

Comment:
The `snapshot info` command fails here if there no clones are created. I think the clone index path gets created on first snapshot clone. If there is no clones created, the index path doesn't exist and snapshot info fails.